### PR TITLE
[PURCHASE-1990] Reply box should not extent when detail box is open

### PR DESCRIPTION
**Pull request by sepans**
Fixes [PURCHASE-1990](https://artsyproduct.atlassian.net/browse/PURCHASE-1990)

<details><summary>screenshots:</summary>
<p>

Note the. `send` button in the bottom of the page

### After:

</details>

**Tip:** In the first commit, I just rearranged the elements a bit so you can just check out the second commit.

In order to fix this, I had to change the `position: fixed` style to flex box. The idea is that instead of sticking the box to the bottom with `position: fixed; bottom: 0;` put the messages and reply box in a flexbox and set the height of messages box to take the entire section. Then move the scrolling to a container inside the 'messages' box.

### CODE REVIEW

File: src/v2/Apps/Conversation/Components/Conversation.tsx [hunk 1]
```diff
@@ -210,21 +211,32 @@ const Conversation: React.FC<ConversationProps> = props => {
...
-    </Flex>
+    </NoScrollFlex>
   )
 }
 
+const MessageContainer = styled(Box)`
+  height: calc(100% - 300px);
+  flex-grow: 1;
+  overflow-y: scroll;
+`
```
Comment (sepans) [inline src/v2/Apps/Conversation/Components/Conversation.tsx:234]: So this `MessageContainer` has its height bound to `100%` minus things above and below it. The scroll happens now happens in this box and the parent (`NoScrollFlex`) doesn't do scroll. `flex-grow: 1` makes the `MessageContainer` to take the entire height available to it so it guarantees that the reply box always appears in the bottom of the page.

File: src/v2/Apps/Conversation/Components/Message.tsx [hunk 2]
```diff
@@ -95,6 +95,7 @@ const Message: React.FC<MessageProps> = props => {
           alignSelf,
         }}
         maxWidth="66.67%"
+        width="fit-content"
```
Comment (sepans) [inline src/v2/Apps/Conversation/Components/Message.tsx:98]: just a little fix I noticed while working on the reply box to limit the width of the message box to its content.

File: src/v2/Apps/Conversation/Components/Reply.tsx [hunk 3]
```diff
@@ -4,19 +4,11 @@ import React, { useRef, useState } from "react"
...
-  ${right};
   border-top: 1px solid ${color("black10")};
-  position: fixed;
   background: white;
-  bottom: 0;
-  left: 375px;
-  ${media.xs`
-    width: 100%;
-    left: 0;
-  `}
```
Comment (sepans) [inline src/v2/Apps/Conversation/Components/Reply.tsx:19]: With the flexbox, we can get rid of all these...

File: src/v2/Apps/Conversation/Routes/Conversation/index.tsx [hunk 4]
```diff
@@ -25,6 +25,9 @@ interface ConversationRouteProps {
 
 const ConstrainedHeightFlex = styled(Flex)`
   height: calc(100vh - 145px);
+  ${media.xs`
+    height: calc(100vh - 55px);
+  `}
```
Comment (sepans) [inline src/v2/Apps/Conversation/Routes/Conversation/index.tsx:30]: There was too much white space on the bottom of the messages page on mobile so had to adjust this.

File: src/v2/Apps/Conversation/Components/Conversation.tsx (hunk 1 above)
Comment (jpotts244) [reply src/v2/Apps/Conversation/Components/Conversation.tsx:235]: Looks like this helps with some of those extra long messages as well. Now you can have access to those messages that were getting hidden.


Screen Recording 2020-07-06 at 12 59 58 PM

**Comment by sepans** [comment]
🙌  Going to merge this to test it on staging.

**Pull request merged**